ert_data/loader.py

```diff
@@ -8,101 +8,215 @@ def data_loader_factory(observation_type):
         See https://github.com/equinor/libres/issues/808
     """
     if observation_type == "GEN_OBS":
-        return load_general_data
+        return _load_general_data
     elif observation_type == "SUMMARY_OBS":
-        return load_summary_data
+        return _load_summary_data
     elif observation_type == "BLOCK_OBS":
-        return load_block_data
+        return _load_block_data
     else:
         raise TypeError("Unknown observation type: {}".format(observation_type))
 
 
-def load_general_data(facade, observation_key, case_name, include_data=True):
-    obs_vector = facade.get_observations()[observation_key]
-    data_key = obs_vector.getDataKey()
+def _extract_data(
```

Comment:
There is a slight API break here, as the returned DataFrame has the observation keys and meta data injected, instead of that information being on the caller side. That is because that context is no longer available on the calling side when we changed from having a single key as input to a list of keys.

Reply:
So we're pretty much adding more information to the output? Should be good.

Reply:
From what I see though this function is not testet outside of the integration test. Are we able to add unit tests for this as well?

Reply:
This has arguably been done now, I added tests testing it though the `loader.data_loader_factory` where just the facade is mocked, do you agree that is enough, or would you like this function called explicitly in a test?

Reply:
I'm pretty happy about the tests now, they look good and from what I can tell tests the functionality of all logic in the `loader` file.